### PR TITLE
Revision of the advanced topics document.

### LINKED ISSUE #341: Review and update Exp/Advanced Topics

### PR DESCRIPTION
**Pull request by jhkrug**
There are 3 TODOs that need attention before being ready to merge. Thanks.

Will close #341 when merged.

### CODE REVIEW

File: docs/explanations/advanced.md [hunk 1]
```diff
@@ -1,158 +1,178 @@
...
-## Prerequisites
+On a Kubernetes cluster, certain services need to be reachable from outside the cluster.
+For Epinio, the `API server` is one of them.
+Certain of the applications deployed with Epinio might also need to be reachable from outside.
 
-There are some components that need to be installed on a Kubernetes cluster before
-Epinio is installed. Epinio helm chart doesn't deploy these components.
+One way to expose services externally is by creating [Ingress resources](https://kubernetes.io/docs/concepts/services-networking/ingress/).
+Ingress resources on their own, have no effect.
+They're just descriptions of what the needed routing.
```
Comment (andreas-kupries) [inline docs/explanations/advanced.md:20]:
```suggestion
They're just descriptions of the needed routing.
```

File: docs/explanations/advanced.md [hunk 2]
```diff
@@ -1,158 +1,178 @@
...
+[Cert-manager documentation](https://cert-manager.io/docs/)
 
-Cert Manager is a Kubernetes controller that generates and renews Certificates needed in order to
-serve the various accessible endpoints over TLS (e.g. the Epinio API server).
+Cert-manager is a Kubernetes controller that generates and renews certificates.
+You need these certificates  to securely serve the endpoints over TLS (for example, the Epinio API server).
 
-Epinio supports various options when it comes to certificate issuers (let's encrypt, private CA, bring your own CA, self signed certs).
-Cert Manager simplifies the way we handle the various different certificate issuers within Epinio.
+Epinio supports options when it comes to certificate issuers (Let's Encrypt, private your own CA, self signed certs).
```
Comment (andreas-kupries) [inline docs/explanations/advanced.md:42]:
```suggestion
Epinio supports options when it comes to certificate issuers (Let's Encrypt, your own private CA, self signed certs).
```

File: docs/explanations/advanced.md [hunk 3]
```diff
@@ -1,158 +1,178 @@
...
-Kubed is installed by Epinio to keep secrets that are needed in more than
-one namespace synced. For example, the image pull secret is needed in every application namespace
+Epinio installs Kubed to keep secrets, needed in more than one namespace, synced.
+For example, the image pull secret is needed in every application namespace
 so that Kubernetes can pull the built application images from the [Container Registry](#container-registry).
 
-Kubed makes sure that if the source secret changes, the copy will change too.
+Kubed makes sure that if the source secret changes, the copies change too.
+
+<!--TODO: Still the case?-->
```
Comment (andreas-kupries) [inline docs/explanations/advanced.md:78]: AFAIK yes. @enrichman ?

File: docs/explanations/advanced.md [hunk 4]
```diff
@@ -1,158 +1,178 @@
...
 
 1. Staging job - pushing the images
-2. Kubernetes - pulling the images when a Deployment is created for the Application
+2. Kubernetes - pulling the images when creating a deployment for the application
 
-Ideally all consumers will communicate with the registry over TLS to keep the communication encrypted.
-Epinio controls the staging job and ensures that whatever CA is used to sign the registry certificate is trusted by it. Achieving the same for Kubernetes however requires configuration that cannot happen from within the cluster, therefore Epinio has no way to ensure that. Theoretically there are 3 options:
+<!--TODO:Did I reword this correctly. I think it needs some better explanation. Who can help here?-->
+All consumers should communicate with the registry using TLS to encrypt communication.
+Epinio controls the staging job and it ensures that iit trusts the CA used to sign the registry certificate.
```
Comment (andreas-kupries) [inline docs/explanations/advanced.md:123]:
```suggestion
Epinio controls the staging job and it ensures that it trusts the CA used to sign the registry certificate.
```

File: docs/explanations/advanced.md [hunk 5]
```diff
@@ -1,158 +1,178 @@
...
+Achieving the same for Kubernetes requires configuration that is impossible in the cluster,
```
Comment (andreas-kupries) [inline docs/explanations/advanced.md:124]:
```suggestion
Achieving the same for Kubernetes requires configuration that is impossible from within the cluster,
```
```suggestion
Achieving the same for Kubernetes requires configuration that is impossible from inside the cluster,
```
I.e. an application like E, running in a pod of the cluster does not and cannot have the permissions to change things so highly relevant to cluster security. Such changes can only be made from the outside, by a cluster operator.

File: docs/explanations/advanced.md [hunk 6]
```diff
@@ -1,158 +1,178 @@
...
+
+There are 3 options:
 
 1. Let the Epinio user manually configure Kubernetes to trust the CA
-2. Use a well-known trusted CA, so there's no configuration needed
-3. Don't encrypt the communication at all
+1. Use a well-known trusted CA, so there's no configuration needed
+1. Don't encrypt the communication at all
+
+<!--TODO:Doesn't sound that great to me. What's happening moving forward, is the current situation accurately described.-->
```
Comment (andreas-kupries) [inline docs/explanations/advanced.md:133]: AFAICT the situation is described correctly. And, AFAIK, we have no way forward out of this.

**Review by andreas-kupries — CHANGES_REQUESTED** [review]

File: docs/explanations/advanced.md (hunk 1 above)
Comment (jhkrug) [reply docs/explanations/advanced.md:20]: Thanks.

File: docs/explanations/advanced.md (hunk 2 above)
Comment (jhkrug) [reply docs/explanations/advanced.md:42]: Thanks.

File: docs/explanations/advanced.md (hunk 5 above)
Comment (jhkrug) [reply docs/explanations/advanced.md:124]: Thanks.

File: docs/explanations/advanced.md (hunk 6 above)
Comment (jhkrug) [reply docs/explanations/advanced.md]: Thanks.

File: docs/explanations/advanced.md (hunk 3 above)
Comment (jhkrug) [reply docs/explanations/advanced.md:81]: Hi @enrichman , Just to be clear i@m asking about whether it's still the case that 'Secret rotation' will be handled by Epinio in the future. Thanks,

Comment (enrichman) [reply docs/explanations/advanced.md:81]: I'm not sure to understand which is the secrets we are referring to, and what kind of rotation. Regarding Kubed and the registry that's probably already the case. For other secrets I don't think that we are going to manage rotations somehow.

**Review by enrichman — DISMISSED** [review]

**Review by andreas-kupries — APPROVED** [review]

**Review by andreas-kupries — APPROVED** [review]

**Pull request merged**